atomic_reactor/plugins/post_push_to_pulp.py

```diff
@@ -176,11 +179,36 @@ def push_tarball_to_pulp(self, image_names, repo_prefix="redhat-"):
                     tags=[tag]
                 )
 
-        top_layer, layers = self._get_tar_metadata(self.filename)
         self.log.info("pulp_repos = %s", pulp_repos)
         self._create_missing_repos(p, pulp_repos, repo_prefix)
 
-        p.upload(self.filename)
+        try:
+            top_layer, layers = self._get_tar_metadata(self.filename)
+            # getImageIdsExist was introduced in rh-dockpulp 0.6+
+            existing_imageids = p.getImageIdsExist(layers)
+            self.log.debug("existing layers: %s" % existing_imageids)
+
+            # Strip existing layers from the tar and repack it
+            remove_layers = [str(os.path.join(x, 'layer.tar')) for x in existing_imageids]
+
+            commands = {'.xz': 'xzcat', '.gz': 'zcat', '.bz2': 'bzcat', '.tar': 'cat'}
+            _, file_extension = os.path.splitext(self.filename)
+            unpacker = commands.get(file_extension, None)
+            self.log.debug("using unpacker %s for extention %s" % (unpacker, file_extension))
+            if unpacker is None:
+                raise Exception("Unknown tarball format: %s" % self.filename)
+
+            with NamedTemporaryFile(prefix='strip_tar_', delete=True) as outfile:
+                cmd = "{0} {1} | tar --delete {2} | gzip - > {3}".format(
+                    unpacker, self.filename, ' '.join(remove_layers), outfile.name)
+                self.log.debug("running %s" % cmd)
+                subprocess.check_call(cmd, shell=True)
+                self.log.debug("uploading %s" % outfile.name)
+                p.upload(outfile.name)
+        except:
```

Comment:
We discussed it [here](https://github.com/projectatomic/atomic-reactor/pull/458#discussion_r59534830) - since we don't know what type of exception may occur (some don't even inherit from ```Exception```) we'd rather have as generic catcher as possible - just to be super-sure that it won't fail